test/test_images/listener/main.go

```diff
@@ -52,8 +82,14 @@ func main() {
 		if event.Extensions()[ceVSphereAPIKey] == "" {
 			logging.FromContext(ctx).Fatalf("CloudEvent extension %q not set", ceVSphereAPIKey)
 		}
-	}); err != nil {
-		logging.FromContext(ctx).Fatalf("receiving events: %v", err)
+
+		count += 1
+
+		if count == numExpectedEvents {
+			logging.FromContext(ctx).Infow("cancelling context: Received expected number of events")
+			cancel()
+			break
+		}
 	}
 
 	if count == 0 {
```

Comment:
This should be unreachable, right?

Reply:
I think what Evan meant is that this `if` branch would never be evaluated. I'd leave the `ctx.Cancel()` as is tbh.